pkg/log/zap/flags.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package zap contains helpers for setting up a new logr.Logger instance
+// using the Zap logging framework.
+
+package zap
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type encoderConfigFunc func(*zapcore.EncoderConfig)
+
+type encoderValue struct {
+	set     bool
+	setFunc func(zapcore.Encoder)
```

Comment:
How about `setFunc`=>`setOptionFunc`? 
`setFunc` alone seems a bit vague.

Reply:
`setFunc` is the callback function that is called when `encoderValue.Set()` is called.

A `setFunc` callback is defined in `BindFlags` so that the `options.Encoder` value can be assigned when the flag is parsed. Just wanted to make sure the intent of the function was clear.